sale_order_engine_data/reports/sale_order_template.xml

```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <template id="report_saleorder_document_engine_data" inherit_id="sale.report_saleorder_document">
+        <xpath expr="//div[@name='expiration_date']" position="after">
+            <div t-if="doc.engine_ref and doc.engine_hours" class="col-auto col-3 mw-100 mb-2" name="engine_data">
```

Comment:
This block should be visible when one of the fields is filled, not both. With this code, if Engine Number is filled but hours net, Engine Number is not shown, and it's not the desired behavior. Please try this instead:
```suggestion
            <div t-if="doc.engine_ref or doc.engine_hours" class="col-auto col-3 mw-100 mb-2" name="engine_data">
```